File: L005/Next-Patterns/pattern10.cpp
#include<iostream>
using namespace std;


void pattern10(int n){

    int outSp = n/2;
   
    int inSp = -1;

    for(int i = 1 ; i <= n ; i++){

        for(int j=1; j <= outSp ; j++){
            cout<<"\t";
        }

        cout<<"*\t";
        
        for(int j=1; j <= inSp ; j++){
            cout<<"\t";
        }

        if(i > 1  && i < n){
            cout<<"*\t";
        }

        if(i <= n/2){

            outSp--;
            
            inSp += 2;
        }

       
        else{

            outSp++;
          
            inSp -= 2;

        }

        cout<<endl;
    }
}
int main(){

    int n;
    cin>>n;

    pattern10(n);

    return 0;
}

File: L005/Next-Patterns/pattern10.java
import java.util.Scanner;

public class pattern10 {

    public static Scanner sc = new Scanner(System.in);

    public static void pattern10_(int n) {

        int outSp = n / 2;

        int inSp = -1;

        for (int i = 1; i <= n; i++) {

            for (int j = 1; j <= outSp; j++) {
                System.out.print("\t");
            }

            System.out.print("*\t");

            for (int j = 1; j <= inSp; j++) {
                System.out.print("\t");
            }

            if (i > 1 && i < n) {
                System.out.print("*\t");
            }

            if (i <= n / 2) {

                outSp--;

                inSp += 2;
            }

            else {

                outSp++;

                inSp -= 2;

            }

            System.out.println();
        }
    }


    public static void main(String []args){

        int n = sc.nextInt();

        pattern10_(n);
    }

}
